apps/src/blockly/customBlocks/googleBlockly/commonBlocks.js

```diff
@@ -2,11 +2,55 @@
  * Defines blocks useful in multiple blockly apps
  */
 
+import {readBooleanAttribute} from '../../utils';
+
+const mutatorProperties = [];
+
 export const blocks = {
   installJoinBlock(blockly) {
     // text_join is included with core Blockly. We register a custom text_join_mutator
     // which adds the plus/minus block UI.
     blockly.Blocks.text_join_simple = blockly.Blocks.text_join;
     blockly.JavaScript.text_join_simple = blockly.JavaScript.text_join;
   },
+  mutationToDom() {
+    var container = Blockly.utils.xml.createElement('mutation');
+    mutatorProperties.forEach(prop => {
+      container.setAttribute(prop, this[prop]);
+    });
+    return container;
+  },
+  domToMutation(mutationElement) {
+    Array.from(mutationElement.attributes).forEach(attr => {
```

Comment:
worth adding a comment explaining what this method does

Reply:
Are you referring to `Array.from` or `domToMutation`? Assuming it's the former, it's because the `.attributes` property returns a live collection which doesn't support array methods like `.forEach`.

https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

Reply:
referring to `domToMutation` generally